src/Workspaces/VisualBasic/Portable/Extensions/ExpressionSyntaxExtensions.vb

```diff
@@ -1288,7 +1288,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions
                     End If
 
                     Dim aliasInfo = semanticModel.GetAliasInfo(name, cancellationToken)
-                    If nameHasNoAlias AndAlso aliasInfo Is Nothing Then
+
+                    ' Don't simplify to predefined type if name is part of a QualifiedName.
+                    ' QualifiedNames can't contain PredefinedTypeNames (although MemberAccessExpressions can).
+                    ' In other words, the left side of a QualifiedName can't be a PredefinedTypeName.
+                    If nameHasNoAlias AndAlso aliasInfo Is Nothing AndAlso Not name.Parent.IsKind(SyntaxKind.QualifiedName) Then
```

Comment:
I won't fix this now since I did not introduce the extra 'if's :) But I agree this code could be better and we should clean it up in the future.